pkg/generate/app/pipeline.go

```diff
@@ -237,6 +237,15 @@ func (p *Pipeline) Objects(accept, objectAccept Acceptor) (Objects, error) {
 		if objectAccept.Accept(build) {
 			objects = append(objects, build)
 		}
+		if p.Build.Source != nil && p.Build.Source.SourceImage != nil && accept.Accept(p.Build.Source.SourceImage) {
+			srcImage, err := p.Build.Source.SourceImage.ImageStream()
```

Comment:
@csrwng did this part so he might know the intent better..   but i wouldn't think it has to be.  What does AsImageStream actually mean/check?  Confirms we were able to create an imagestream for the image reference?